Penteract/Assets/Scripts/Onimaru.cpp

```diff
@@ -82,37 +85,33 @@ void Onimaru::Blast() {
 							if (meleeScript) meleeScript->EnableBlastPushBack();
 							else if (rangedScript) rangedScript->EnableBlastPushBack();
 						}
-					}
-					else {
+					} else {
 						if (meleeScript) {
 							if (!meleeScript->IsBeingPushed()) meleeScript->DisableBlastPushBack();
-						}
-						else if (rangedScript) {
+						} else if (rangedScript) {
 							if (!rangedScript->IsBeingPushed()) rangedScript->DisableBlastPushBack();
 						}
 					}
 				}
 			}
 		}
-	}
-	else {
+	} else {
 		currentBlastDuration += Time::GetDeltaTime();
 		if (currentBlastDuration >= 1.8f) OnAnimationSecondaryFinished(); // Temporary hack
 	}
 }
 
 void Onimaru::PlayAnimation() {
 	if (!compAnimation) return;
-	if (ultimateInUse || !isAlive) return; //Ultimate will block out all movement and idle from happening
+	if (ultimateOn || !isAlive) return; //Ultimate will block out all movement and idle from happening

```

Comment:
I thought that the purpose of having a PlayAnimation function was that this function is the one that sends all animation triggers. This check makes me believe that in Onimaru is not being implemented like this.

Reply:
Most of the triggers sent it to the state machines are done in the _Update()_ method instead in the _PlayAnimation()_

Reply:
In Fang.cpp all the send triggers for starting the abilities are done in the PlayAnimation function, I think that in Onimaru should be there as well to make the code more cohesive and comprehensive.

Reply:
Of course

Reply:
On Fang shooting animation is handled from the Shooting method, thus not all send triggers are handled through PlayAnimation method, should this be changed as well?

Reply:
Yes. I thought that there were done in PlayAnimation, but after reading the code again, I saw that the ones in there where only the dying transitions when shooting. I can do it when I update the model with the new animations if you have another tasks pending

Reply:
Taking into account that his changes are not related to this, I would do it in another PR.

Reply:
I still believe the code will be 100 times more confusing and it will take some big changes to make all animations sendTrigger on that single method, some functionalities take into account the fact that something was happening and no longer is, if this is considered necesary I recommend making it a separate PR, and maybe even someone else other than me with a clearer approach, since it just seems confusing to me

Reply:
I am not sayign all the send trigger have to be in the PlayAnimation function, just the ones that do not depend on anything else, such as the movement, dying and starting abilities. Obviously, the ones sent when an animation finishes have more sense in the OnAnimationFinished function

Reply:
The rework of how the animations are sent is not a high priority and has nothing to do with this PR. The changes work and are ready to be uploaded to develop. Let's not block this task anymore.